server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -65,6 +65,101 @@ impl From<messageattempt::Model> for MessageAttemptOut {
     }
 }
 
+// TODO: Mostly similar to `v1::endpoints::message::MessageOut, is there some way to reduce
+// duplication cleanly?
+
+/// A model containing information on a given message plus additional fields on the last attempt for
+/// that message.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ModelOut, FromQueryResult)]
+#[serde(rename_all = "camelCase")]
+struct AttemptedMessageOut {
+    event_type: EventTypeName,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<MessageUid>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    channels: Option<EventChannelSet>,
+    payload: serde_json::Value,
+    id: MessageId,
+    timestamp: DateTimeWithTimeZone,
+    status: MessageStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    next_attempt: Option<DateTimeWithTimeZone>,
+}
+
+impl AttemptedMessageOut {
+    pub fn from_message_and_destination(
+        message: message::Model,
+        destination: messagedestination::Model,
+    ) -> Self {
+        Self {
+            event_type: message.event_type,
+            event_id: message.uid,
+            channels: message.channels,
+            payload: message.payload,
+            id: message.id,
+            timestamp: message.created_at,
+            status: destination.status,
+            next_attempt: destination.next_attempt,
+        }
+    }
+}
+
+/// Additional parameters (besides pagination) in the query string for the "List Attempted Messages"
+/// enpoint.
+#[derive(Debug, Deserialize, Validate)]
+pub struct ListAttemptedMessagesQueryParameters {
+    #[validate]
+    channel: Option<EventChannel>,
+    status: Option<MessageStatus>,
+}
+
+/// GET /api/v1/app/:app_id/enpoint/:endp_id/msg/
+/// Fetches a list of [`AttemptedMessageOut`]s associated with a given app and endpoint.
+async fn list_attempted_messages(
+    Extension(ref db): Extension<DatabaseConnection>,
+    pagination: ValidatedQuery<Pagination<MessageId>>,
+    ValidatedQuery(ListAttemptedMessagesQueryParameters { channel, status }): ValidatedQuery<
+        ListAttemptedMessagesQueryParameters,
+    >,
+    Path((_app_id, endp_id)): Path<(ApplicationId, EndpointId)>,
+    AuthenticatedApplication {
+        permissions: _,
+        app: _,
+    }: AuthenticatedApplication,
+) -> Result<Json<ListResponse<AttemptedMessageOut>>, Error> {
+    let limit = pagination.limit;
+    let iterator = pagination.iterator.clone();
+
+    let mut msg_and_dest = messagedestination::Entity::secure_find_by_endpoint(endp_id)
+        .order_by_desc(messagedestination::Column::MsgId)
+        .limit(limit + 1)
+        .join(
+            sea_orm::JoinType::Join,
+            messagedestination::Entity::has_one(message::Entity).into(),
+        );
+
+    if let Some(iterator) = iterator {
+        msg_and_dest = msg_and_dest.filter(messagedestination::Column::MsgId.gt(iterator));
```

Comment:
```suggestion
        msg_and_dest = msg_and_dest.filter(messagedestination::Column::MsgId.lt(iterator));
```
Watch out for these things, we order by descending order, so it should be `lt`, not `gt`.